src/cockpit/packages.py

```diff
@@ -234,21 +242,29 @@ def load_file(self, filename: str) -> Document:
 
         return Document(path.open('rb'), content_type, content_encoding, content_security_policy)
 
-    def load_translation(self, locales: List[str]) -> Document:
+    def load_translation(self, path: str, locales: List[str]) -> Document:
         self.ensure_scanned()
-        assert self.translations is not None
+        if path == 'po.js':
+            translations = self.translations
+        elif self.manifest_translations:
```

Comment:
That looks wrong.  Should rather have a `not` there, no?  But I'd actually change the bodies instead.

I still like the idea of adding a level of indirection on the translations by filename, so you could lookup

`self.translations['po.js']['de']` or `self.translations['po.manifest.js']['de']`

Then you could implement the fallback as a one-time check from `ensure_scanned()` — if `self.translations['po.manifest.js']` is empty then you set it equal to `self.translations['po.js']` and don't have to think about it anymore.